File: app/src/main/java/com/qianlei/jiaowu/ui/fragment/score/ScoreAdapter.java
package com.qianlei.jiaowu.ui.fragment.score;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.qianlei.jiaowu.R;
import com.qianlei.jiaowu.entity.Score;
import com.qianlei.jiaowu.utils.ScoreUtil;

import java.util.List;

/**
 * 成绩适配器
 *
 * @author qianlei
 */
public class ScoreAdapter extends RecyclerView.Adapter<ScoreAdapter.ScoreViewHolder> {
    private List<Score> scoreList;

    ScoreAdapter(List<Score> scoreList) {
        this.scoreList = scoreList;
    }

    @NonNull
    @Override
    public ScoreViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.score_item, parent, false);

        return new ScoreViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ScoreViewHolder holder, int position) {
        if (position == 0) {
            //第一个显示GPA信息
            holder.scoreText.setText(ScoreUtil.getGpa(scoreList));
            holder.scoreNameText.setText(R.string.gpa);
            return;
        }
        position--;
        holder.scoreText.setText(scoreList.get(position).getScore());
        holder.scoreNameText.setText(scoreList.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return scoreList.size() + 1;
    }

    static class ScoreViewHolder extends RecyclerView.ViewHolder {
        TextView scoreNameText;
        TextView scoreText;

        ScoreViewHolder(@NonNull View itemView) {
            super(itemView);
            scoreNameText = itemView.findViewById(R.id.score_name);
            scoreText = itemView.findViewById(R.id.score_score);
        }

    }
}

File: app/src/main/java/com/qianlei/jiaowu/ui/fragment/score/ScoreAdapter.kt
package com.qianlei.jiaowu.ui.fragment.score

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.qianlei.jiaowu.R
import com.qianlei.jiaowu.entity.Score
import com.qianlei.jiaowu.ui.fragment.score.ScoreAdapter.ScoreViewHolder
import com.qianlei.jiaowu.utils.ScoreUtil

/**
 * 成绩适配器
 *
 * @author qianlei
 */
class ScoreAdapter internal constructor(private val scoreList: List<Score>?) : RecyclerView.Adapter<ScoreViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ScoreViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.score_item, parent, false)
        return ScoreViewHolder(view)
    }

    override fun onBindViewHolder(holder: ScoreViewHolder, position: Int) {
        if (scoreList == null) {
            return
        }
        if (position == 0) { //第一个显示GPA信息
            holder.scoreText.text = ScoreUtil.getGpa(scoreList)
            holder.scoreNameText.setText(R.string.gpa)
            return
        }
        holder.scoreText.text = scoreList[position - 1].score
        holder.scoreNameText.text = scoreList[position - 1].name
    }

    override fun getItemCount(): Int {
        if (scoreList == null) {
            return 1
        }
        return scoreList.size + 1
    }

    class ScoreViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var scoreNameText: TextView = itemView.findViewById(R.id.score_name)
        var scoreText: TextView = itemView.findViewById(R.id.score_score)
    }

}